### PR TITLE
oboetester: try to repro routing crash

### PR DESCRIPTION
**Pull request by philburk**
This test tries to trigger a routing change
while long duration callbacks are playing.
This can cause the callback to write into a dead buffer and cause a SIGSEGV.

See b/274815060

### CODE REVIEW

File: apps/OboeTester/app/src/main/cpp/TestRoutingCrash.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 The Android Open Source Project
```
Comment (flamme) [inline apps/OboeTester/app/src/main/cpp/TestRoutingCrash.cpp:2]: 2023?

File: apps/OboeTester/app/src/main/cpp/TestRoutingCrash.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 The Android Open Source Project
```
Comment (flamme) [inline apps/OboeTester/app/src/main/cpp/TestRoutingCrash.h:2]: 2023?

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestRouteDuringCallbackActivity.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 The Android Open Source Project
```
Comment (flamme) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestRouteDuringCallbackActivity.java:2]: 2023?

**Review by flamme — COMMENTED** [review]
Thanks Phil for the PR!

The restoring the track seems pretty relies on the timing to making a crash. Is it possible to have a call from JNI to JAVA when receiving a data callback to force routing change?

File: apps/OboeTester/app/src/main/res/layout/activity_routing_crash.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        android:layout_height="wrap_content"
+        android:orientation="vertical">
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:fontFamily="monospace"
+            android:gravity="bottom"
+            android:text="
+Try to cause a crash by changing the\n
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/res/layout/activity_routing_crash.xml:21]: Put into strings.xml?

File: apps/OboeTester/app/src/main/cpp/TestRoutingCrash.cpp (hunk 1 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/cpp/TestRoutingCrash.cpp:2]: Done

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestRouteDuringCallbackActivity.java (hunk 3 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestRouteDuringCallbackActivity.java:2]: Done

File: apps/OboeTester/app/src/main/res/layout/activity_routing_crash.xml (hunk 4 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/res/layout/activity_routing_crash.xml:21]: Done

**Pull request merged**